### PR TITLE
changes for ros2 - rebased

### PR DESCRIPTION
**Pull request by ijnek**
Rebased #16 onto most recent master, with some changes.

Additional changes:
* Change the executable to a ROS2 component (using ros2 component magic!). This allows the node to be reused in many other ways than just an executable. Same syntax `ros2 run humanoid_base_footprint base_footprint` can still be used to run the node as a standalone executable. The existing launch file works fine too.
* Bunch of styling fixes to pass CI
* Moving some #includes from .hpp to .cpp
* Change .h to .hpp, as that's recommended ROS2 style.
* Changes license to Apache License V2.0 (ROS2 Recommended) - this is an important change, if you prefer keeping the old license, we need to look into getting ament tests passing with it.

### CODE REVIEW

File: .rdmanifest [hunk 1]
```diff
@@ -5,7 +5,6 @@ check-presence-script: '#!/bin/bash
   test -d $BITBOTS_CATKIN_WORKSPACE/src/humanoid_base_footprint'
 depends:
 - bitbots_docs
```
Comment (Flova) [inline .rdmanifest:7]: bitbots_docs is probably not needed anymore for ros-sports

Comment (ijnek) [reply .rdmanifest:7]: I'd like to remove `.rdrmanifest` as it is specific to bit-bots. Can @Flova or @SammyRamone remove this file without breaking bit-bots's CI.

Comment (Flova) [reply .rdmanifest:7]: I have never worked with our CI. Maybe @SammyRamone, @ftsell or @timonegk.

Comment (SammyRamone) [reply .rdmanifest:7]: I also have never worked with the CI. Only @ftsell and @timonegk know about it

**Review by SammyRamone — APPROVED** [review]

File: .rdmanifest (hunk 1 above)
Comment (ijnek) [reply .rdmanifest:7]: Discussion moved to #25, not going to remove for this PR.

**Pull request merged**